### PR TITLE
[NSURL] Implementation of '.checkResourceIsReachable'

### PR DESCRIPTION
**Pull request by naithar**
Implemented and renamed method for `NSURL` to match Darwin's version.
Added method `. checkResourceIsReachable` to `URL` that matches Darwin's Foundation.

### CODE REVIEW

File: Foundation/NSURL.swift [hunk 1]
```diff
@@ -610,8 +610,28 @@ open class NSURL : NSObject, NSSecureCoding, NSCopying {
     */
     /// - Experiment: This is a draft API currently under consideration for official import into Foundation as a suitable alternative
     /// - Note: Since this API is under consideration it may be either removed or revised in the near future
-    open func checkResourceIsReachable() throws -> Bool {
-        NSUnimplemented()
+    open func checkResourceIsReachableAndReturnError(_ error: UnsafeMutablePointer<NSError?>?) -> Bool {
```
Comment (parkera) [inline Foundation/NSURL.swift]: Did you put back the error argument here for compatibility with Darwin? There it's an autoreleased error, which we don't have on Linux. For this method I'm ok with the divergent API for now.

Comment (naithar) [inline Foundation/NSURL.swift]: Yes, I did it for compatibility with Darwin. I didn't know about the issue (i guess i've just missed it). I'll revert this change.

**Comment by naithar** [comment]
Updated `NSURL` class. @parkera please review.

**Comment by parkera** [comment]
@swift-ci please test and merge

**Comment by naithar** [comment]
"ninja: error: unknown target 'swift-benchmark-linux-x86_64"
Seems like unrelated issue.

**Comment by parkera** [comment]
@shahmishal @erg do you know what that error means?

**Comment by shahmishal** [comment]
This PR should fix this issue https://github.com/apple/swift/pull/7931

**Comment by parkera** [comment]
@swift-ci please test and merge

**Pull request merged**